### PR TITLE
Mac OSX Yosemite+ support via pfctl and dnctl

### PR DESCRIPTION
**Pull request by mefellows**
Adds support for Mac OSX Yosemite via the `pfctl` and `dnctl` commands. As part of the implementation, I've removed the check for OSX version and instead replaced it with tests for the 2 known firewall implementations (namely `pfctl` and `ipfw`).

If this makes it in, we may want to consider updating the OSX stanza in the README.

Nice little project, btw :ok_hand:  (although the name has less effect over here in Australia).

### CODE REVIEW

**Comment by tylertreat** [comment]
This is excellent! Will try to get through this PR later today.

File: throttler/throttler.go [hunk 1]
```diff
@@ -144,12 +146,17 @@ func Run(cfg *Config) {
 	}
 }
 
-func osxVersionSupported() bool {
+func osxFirewall() string {
```
Comment (tylertreat) [inline throttler/throttler.go]: This doesn't appear to be called from anywhere?

**Comment by Term1nal** [comment]
Awesome!

File: throttler/throttler.go (hunk 1 above)
Comment (mefellows) [inline throttler/throttler.go]: Good pickup - I originally modified it but now i've removed it altogether.

**Comment by mefellows** [comment]
@tylertreat I have removed the unused function (OSXFirewall()`) and squashed the commit.

**Comment by tylertreat** [comment]
Thanks!

**Pull request merged**